micromasters/models.py

```diff
@@ -0,0 +1,83 @@
+"""
+Classes related to models for MicroMasters
+"""
+
+from django.contrib.auth.models import User
+from django.contrib.postgres.fields import JSONField
+from django.db import transaction
+from django.db.models import (
+    DateTimeField,
+    ForeignKey,
+    Model,
+    SET_NULL,
+)
+
+
+class AuditModel(Model):
+    """An abstract base class for audit models"""
+
+    acting_user = ForeignKey(User, null=True, on_delete=SET_NULL)
+    created_on = DateTimeField(auto_now_add=True)
+    updated_on = DateTimeField(auto_now=True)
+    data_before = JSONField(blank=True, null=True)
+    data_after = JSONField(blank=True, null=True)
+
+    class Meta:
+        abstract = True
+
+    @classmethod
+    def get_related_field_name(cls):
+        """
+        Returns:
+            str: A field name which links the Auditable model to this model
+        """
+        raise NotImplementedError
+
+
+class AuditableModel(Model):
+    """An abstract base class for auditable models"""
+
+    class Meta:
+        abstract = True
+
+    def to_dict(self):
+        """
+        Returns:
+            dict:
+                A serialized representation of the model object
+        """
+        raise NotImplementedError
```

Comment:
i feel like `serialize_mobel_object(self)` is a reasonable base implementation for this